### PR TITLE
adding zadatak 12

### PR DESCRIPTION
**Pull request by aadma**

### CODE REVIEW

File: zadaca12/zadaca11/zadaca11/UserViewController.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    var user: User!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.delegate = self
+        tableView.dataSource = self
+         user = UserProfile.shared.userProfile
+        nameLabel.text = user.name
+        let rand = Int(arc4random_uniform(2) + 1)
+        if rand == 2{
+        icon.image = #imageLiteral(resourceName: "icon1")
```
Comment (irfanG) [inline zadaca12/zadaca11/zadaca11/UserViewController.swift]: nije ti dobar alignment - treba biti uvuceno za jedan tab

File: zadaca12/zadaca11/zadaca11/UserViewController.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }else{
```
Comment (irfanG) [inline zadaca12/zadaca11/zadaca11/UserViewController.swift]: dodaj malo space `} else {`

File: zadaca12/zadaca11/zadaca11/UserViewController.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            cell.detailTextLabel?.text = user.name
+        case 1:
+            cell.textLabel?.text = "Username"
+            cell.detailTextLabel?.text = user.username
+        case 2:
+            cell.textLabel?.text = "Email"
+            cell.detailTextLabel?.text = user.email
+        case 3:
+            cell.textLabel?.text = "Address"
+            cell.detailTextLabel?.text = (user.address?.street!)! + ", " + (user.address?.city!)!
```
Comment (irfanG) [inline zadaca12/zadaca11/zadaca11/UserViewController.swift]: sta ako je city = nil ili street = nil

File: zadaca12/zadaca11/zadaca11/UsersViewController.swift [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    
+    func putUsersOnMap(){
+        
+        for user in users{
+            let annotation = MKPointAnnotation()
+            let lat = user.address?.geo?.lat
+            let lng = user.address?.geo?.lng
+            let coordinate = CLLocationCoordinate2DMake(Double(lat!)!, Double(lng!)!)
```
Comment (irfanG) [inline zadaca12/zadaca11/zadaca11/UsersViewController.swift]: sta ako se lat, lng  iz string ne moze castati u Double

File: zadaca12/zadaca11/zadaca11/UsersViewController.swift [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+    @IBAction func sort(_ sender: UIBarButtonItem) {
+        guard users.count > 0 else{
+            return
+        }
+        sortUsersByDistance()
+    }
+    
+    func displayUser(){
+        performSegue(withIdentifier: "showUserSegue", sender: self)
+
```
Comment (irfanG) [inline zadaca12/zadaca11/zadaca11/UsersViewController.swift]: visak space

File: zadaca12/zadaca11/zadaca11/Base.lproj/Main.storyboard [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+                                <constraints>
+                                    <constraint firstAttribute="width" constant="288" id="Cc8-n7-WdT"/>
+                                    <constraint firstAttribute="height" constant="21.5" id="lT2-dS-N7B"/>
+                                </constraints>
+                                <fontDescription key="fontDescription" type="boldSystem" pointSize="18"/>
+                                <nil key="textColor"/>
+                                <nil key="highlightedColor"/>
+                            </label>
+                            <tableView clipsSubviews="YES" contentMode="scaleToFill" alwaysBounceVertical="YES" dataMode="prototypes" style="plain" rowHeight="44" sectionHeaderHeight="28" sectionFooterHeight="28" translatesAutoresizingMaskIntoConstraints="NO" id="lAD-qM-mPB">
+                                <color key="backgroundColor" white="1" alpha="1" colorSpace="calibratedWhite"/>
```
Comment (irfanG) [inline zadaca12/zadaca11/zadaca11/Base.lproj/Main.storyboard]: nisi bas sve odradio ovdje sto je trazeno: gradient, linije... provjeri jos jednom
screen shot 2016-10-29 at 08 38 17

**Comment by irfanG** [comment]
eh da, imas jos jednu greskicu na constraintima : ```which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x610000098d30 MKMapView:0x7ff98cd07780.height == 333   (active)>",
    "<NSLayoutConstraint:0x61000009d100 UILayoutGuide:0x610000197690'Edge Insets'.top == MKMapView:0x7ff98cd07780.top + 72   (active)>",
    "<NSLayoutConstraint:0x61000009d330 UILayoutGuide:0x610000197690'Edge Insets'.bottom == MKMapView:0x7ff98cd07780.bottom - 272   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x61000009d330 UILayoutGuide:0x610000197690'Edge Insets'.bottom == MKMapView:0x7ff98cd07780.bottom - 272   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
```

File: zadaca12/zadaca11/zadaca11/Base.lproj/Main.storyboard (hunk 6 above)
Comment (aadma) [inline zadaca12/zadaca11/zadaca11/Base.lproj/Main.storyboard]: za gradient su potrebne 2 boje a ovde je navedena samo jedna...mnogo je primjera za linear gradient na netu ali rijetko koji za radial...i ono sta sam nasao nisam uspio razumijeti kako se implementira

**Comment by aadma** [comment]
U vezi greške na konstraintima...meni ovaj debbuging nije jasan nikako i najdalje sto sam uspio učinit prateći uputsvo:
https://developer.apple.com/library/content/documentation/UserExperience/Conceptual/AutolayoutPG/DebuggingTricksandTips.html#//apple_ref/doc/uid/TP40010853-CH21-SW1 i neke primjere sa stackoverflow jeste da smanjim priority na map width i onda nestane ova greška ali se pojavi novi problem   tj poveča se map height sam od sebe u runtime-u

**Pull request merged**